includes/data-stores/class-wc-customer-download-data-store.php

```diff
@@ -61,12 +116,28 @@ public function create( &$download ) {
 			apply_filters( 'woocommerce_downloadable_file_permission_format', $format, $data )
 		);
 
-		if ( $result ) {
-			$download->set_id( $wpdb->insert_id );
-			$download->apply_changes();
+		return $result ? $wpdb->insert_id : false;
+	}
+
+	/**
+	 * Adjust a date value to be inserted in the database.
+	 *
+	 * @param mixed $date The date value. Can be a WC_DateTime, a timestamp, or anything else that "date" recognizes.
+	 * @return string The date converted to 'Y-m-d' format.
+	 * @throws Exception The passed value can't be converted to a date.
+	 */
+	private function adjust_date_for_db( $date ) {
+		if ( 'WC_DateTime' === get_class( $date ) ) {
+			$date = $date->getTimestamp();
 		}
 
-		do_action( 'woocommerce_grant_product_download_access', $data );
+		$adjusted_date = date( 'Y-m-d', $date ); // phpcs:ignore WordPress.DateTime.RestrictedFunctions.date_date
+
+		if ( $adjusted_date ) {
+			return $adjusted_date;
+		}
+
+		throw new Exception( "I don't know how to get a date from a " . is_object( $date ) ? get_class( $date ) : gettype( $date ) );
```

Comment:
Does this not need to be translatable?

Reply:
My understanding was that since it's an exception message it should rarely be displayed and if it is, the viewer will probably be a developer. But anyway, I see that our convention in the code is to translate the exception messages so I'll do the same.